tilavarauspalvelu/api/graphql/types/reservation/serializers/update_serializers.py

```diff
@@ -40,7 +41,7 @@ def validate(self, data, prefill_from_profile=False):
             msg = f"Setting the reservation state to '{getattr(new_state, 'value', new_state)}' is not allowed."
             raise ValidationErrorWithCode(msg, ValidationErrorCodes.STATE_CHANGE_NOT_ALLOWED)
 
-        data = super().validate(data, prefill_from_profile)
+        data = super().validate(data)
```

Comment:
I think the behavior would change here, since `ReservationUpdateSerializer` inherits from `ReservationCreateSerializer`, which has `prefill_from_profile=True` by default, while here we would pass `False`. This would cause the reserve info to change to the profile info on update, which could be different if the user changed it before reservation confirmation.

Interestingly, `prefill_from_profile=False` was also included in the `ReservationConfirmSerializer`, but not passed down to to the superclass. Maybe this does work accidentally, since most handlers don't have permissions to read other user's profile info, so the prefill wouldn't override anything.

In any case, to fix this we would need to remove the inheritance between the two serializer and migrate the serializer methods to actions for resuability. This is a bit of work though, so maybe you can add some smaller fix before that.

Reply:
Excellent point!
Resolved this by creating a base class for these two serializers, and separating the prefill logic to only the create serializer.

Reply:
Yep, this will do.